src/big/big_decimal.cr

```diff
@@ -0,0 +1,241 @@
+require "big_int"
+
+# A `BigDecimal` represents arbitrary precision decimals.
+#
+# It is internally represented by a pair of BigInt and UInt64: value and scale.
+# Value contains the actual value, and scale tells the decimal point place.
+# e.g. value=1234, scale=2 => 12.34
+#
+# The general idea and some of the arithmetic algorithms were adapted from
+# the MIT/APACHE -licensed https://github.com/akubera/bigdecimal-rs
+
+class InvalidBigDecimalException < Exception
+end
+
+struct BigDecimal
+  ZERO = BigInt.new(0)
+  TEN  = BigInt.new(10)
+  @@max_div_iterations = 100_u64
+
+  include Comparable(BigDecimal)
+  include Comparable(Int)
+  include Comparable(String)
+
+  private property value : BigInt
+  private property scale : UInt64
+  getter value, scale
+
+  # Set maximum iterations used in division operation. This implicitly
+  # defines the maximum precision of divisions in cases where the
+  # division is not exact.
+  def self.set_max_div_iterations(x : Int)
+    @@max_div_iterations = x.to_u64
+  end
+
+  # Create a new `BigDecimal` from a String.
+  #
+  # Allows only valid number strings with an optional negative sign.
+  def initialize(s : String)
```

Comment:
@MakeNowJust Quoted remark applies to documentation, not code-review comments, mind you.